Feature string caching

mehmetcanay opened the pull request:
closes #58

tiadams commented on datastew/embedding.py:
```diff
@@ -1,18 +1,22 @@
 
 class EmbeddingModel(ABC):
-    def __init__(self, model_name: str):
+    def __init__(self, model_name: str, cache_size: int = 10000):
```
Can we add the caching option as a default param? I'd even default it to False since for small applications we will not need it. Makes sense to activiate it later e.g. for index.

mehmetcanay replied:
Done!

tiadams replied:
Looks like it will use caching regardless of the param as of rn

tiadams replied:
Should only initiate/use the cache if self.cache=True

tiadams commented on tests/test_embedding.py:
```diff
@@ -35,6 +37,40 @@ def test_sanitization(self):
-        self.assertListEqual(embedding1.tolist(), embedding2.tolist())
+        self.assertSequenceEqual(embedding1, embedding2)
+
+    def test_caching_get_embedding(self):
```
Could you add a test for cache=True / cache= False?

Getting a uncached embedding should be marginally faster when cache=False since cache does not need to be checked at all

mehmetcanay commented:
@tiadams I implemented the test and also refactored the code. Now it should skip the cache check if cache is set to False. I could not find more similarities between cached and uncached logics to move outside of the "if-else" block. Please let me know if you see any.

tiadams commented:
Thanks for the changes, looks good to me!
